File: src/graph/model/ValueChange.java
package graph.model;

//import graph.model.IGraphModel.AtomicGraphModelChange;

public class ValueChange extends AtomicGraphModelChange
{

	/**
	 *
	 */
	protected Object cell, value, previous;

	/**
	 * 
	 */
	public ValueChange()
	{
		this(null, null, null);
	}

	/**
	 * 
	 */
	public ValueChange(GraphModel model, Object cell, Object value)
	{
		super(model);
		this.cell = cell;
		this.value = value;
		this.previous = this.value;
	}

	/**
	 * 
	 */
	public void setCell(Object value)
	{
		cell = value;
	}

	/**
	 * @return the cell
	 */
	public Object getCell()
	{
		return cell;
	}

	/**
	 * 
	 */
	public void setValue(Object value)
	{
		this.value = value;
	}

	/**
	 * @return the value
	 */
	public Object getValue()
	{
		return value;
	}

	/**
	 * 
	 */
	public void setPrevious(Object value)
	{
		previous = value;
	}

	/**
	 * @return the previous
	 */
	public Object getPrevious()
	{
		return previous;
	}

	/**
	 * Changes the root of the model.
	 */
	public void execute()
	{
		value = previous;
		previous = ((GraphModel) model)._valueForCellChanged(cell,
				previous);
	}

}

File: lib/model/value_change.dart
part of graph.model;

class ValueChange extends AtomicGraphModelChange {

  Object cell, value, previous;

  ValueChange([GraphModel model = null, Object cell = null,
              Object value = null]) : super(model) {
    this.cell = cell;
    this.value = value;
    this.previous = this.value;
  }

  void setCell(Object value) {
    cell = value;
  }

  /**
   * Returns the cell.
   */
  Object getCell() {
    return cell;
  }

  void setValue(Object value) {
    this.value = value;
  }

  /**
   * Returns the value.
   */
  Object getValue() {
    return value;
  }

  void setPrevious(Object value) {
    previous = value;
  }

  /**
   * Returns the previous.
   */
  Object getPrevious() {
    return previous;
  }

  /**
   * Changes the root of the model.
   */
  void execute() {
    value = previous;
    previous = (model as GraphModel)._valueForCellChanged(cell, previous);
  }

}
